ipld/kamt/src/ext.rs

```diff
@@ -145,13 +145,13 @@ impl Extension {
     }
 
     /// Build an extension from a prefix of some hashed bits, starting from however
-    /// far it has been consumed so far, taking the next `consumed` bits.
-    pub fn from_bits(bits: &mut HashBits, mut consumed: u8) -> Result<Extension, Error> {
+    /// far it has been consumed so far, taking the next `length` bits.
+    pub fn from_bits(bits: &mut HashBits, mut length: u32) -> Result<Extension, Error> {
         let mut builder = ExtensionBuilder::new();
-        while consumed > 0 {
-            let i = min(consumed, 8);
+        while length > 0 {
+            let i = min(length, 8);
             let n = bits.next(i as u32)? as u8;
-            consumed -= i;
+            length -= i;
```

Comment:
This will probably be _fewer_ instructions in wasm. It may be slower (wall clock) on x86, but that's not too important.

Reply:
Sorry, what will be fewer instructions? `u32`? I can't see any change in the code :thinking:

Reply:
Yes, u32.

Reply:
Not really? That just leads to more casting (and potential errors) and won't improve performance.